x-pack/plugins/actions/server/sub_action_framework/executor.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger } from '@kbn/core/server';
+import { ActionsConfigurationUtilities } from '../actions_config';
+import { ExecutorType } from '../types';
+import { ExecutorParams, SubActionConnectorType } from './types';
+
+const isFunction = (v: unknown): v is Function => {
+  return typeof v === 'function';
+};
+
+export const buildExecutor = <Config, Secrets>({
+  configurationUtilities,
+  connector,
+  logger,
+}: {
+  connector: SubActionConnectorType<Config, Secrets>;
+  logger: Logger;
+  configurationUtilities: ActionsConfigurationUtilities;
+}): ExecutorType<Config, Secrets, ExecutorParams, unknown> => {
+  return async ({ actionId, params, config, secrets, services }) => {
+    const subAction = params.subAction;
+    const subActionParams = params.subActionParams;
+
+    const service = new connector.Service({
+      connector: { id: actionId, type: connector.id },
+      config,
+      secrets,
+      configurationUtilities,
+      logger,
+      services,
+    });
+
+    const subActions = service.getSubActions();
+
+    if (subActions.size === 0) {
+      throw new Error('You should register at least one subAction for your connector type');
+    }
+
+    const action = subActions.get(subAction);
+
+    if (!action) {
+      throw new Error('Sub action not registered');
```

Comment:
Not clear to me if we will end up getting the subaction name logged as well - guessing we won't.  And guessing that we will probably want that.  I suspect the connector name and/or id will end up showing up though, from the actions framework catching the error.  

I noticed this message in the functional test with no subaction name in the message, which led me here.